server/src/main/java/org/elasticsearch/action/search/SearchShardsGroup.java

```diff
@@ -45,13 +54,20 @@ public SearchShardsGroup(ShardId shardId, List<String> allocatedNodes, boolean s
         this.shardId = oldGroup.getShardId();
         this.allocatedNodes = Arrays.stream(oldGroup.getShards()).map(ShardRouting::currentNodeId).toList();
         this.skipped = false;
+        // This value is specific to resharding feature and this code path is specific to CCS
+        // involving 8.x remote cluster.
+        // We don't currently expect resharding to be used in such conditions so it's unset.
+        this.reshardSplitShardCountSummary = SplitShardCountSummary.UNSET;
```

Comment:
I can open a PR to remove this BWC after your PR, as it is no longer needed in 9.x.

Reply:
Oh nice, we don't allow CCS between majors?